### PR TITLE
Store also datetime for failed tests

### PR DESCRIPTION
**Pull request by marusak**
This will fail with the current DB. We can either drop it, or migrate it. @martinpitt WDYT?

### CODE REVIEW

File: store-failures [hunk 1]
```diff
@@ -78,10 +78,20 @@ def main():
...
         logging.debug("scanning log %s", raw_log)
 
+        # Parse time from log file name. Wrap in try/except block if the url would be unexpected
+        time = "now"
+        try:
+            parts = url.split("/")[-2].split("-")
+            day = parts[2]
+            hours = parts[3]
+            time = "{0}-{1}-{2} {3}:{4}:{5}".format(day[0:4], day[4:6], day[6:], hours[0:2], hours[2:4], hours[4:])
+        except:
```
Comment (martinpitt) [inline store-failures:88]: No blanket exceptions, please; I think what we can reasonably expect is `IndexError`.

Also, I'd rather just store it as a unix timestamp, as a number. Makes age selection/calculation much easier. You can use strftime() for that.

File: store-failures [hunk 2]
```diff
@@ -94,6 +104,7 @@ def init_db(cursor):
                      project TEXT,
                      context TEXT,
                      url TEXT,
+                     time DATETIME,
```
Comment (martinpitt) [inline store-failures:107]: ... and use TIMESTAMP (not much difference in SQLite, but better be explicit)

File: store-failures (hunk 1 above)
Comment (marusak) [reply store-failures]: changed

**Comment by martinpitt** [comment]
Idea from IRC discussion while we are changing the DB format: change `TestRuns.id INTEGER` → `TestRuns.ref TEXT`, and store the full SHA. Then the ID is something meaningful, and we can easily re-import/recreate the DB from that list.

**Pull request merged**